hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -924,11 +925,11 @@ private void startCommit(String instantTime, String actionType, HoodieTableMetaC
         "Found pending restore in active timeline. Please complete the restore fully before proceeding. As of now, "
             + "table could be in an inconsistent state. Pending restores: " + Arrays.toString(inflightRestoreTimeline.getInstantsAsStream()
             .map(instant -> instant.getTimestamp()).collect(Collectors.toList()).toArray()));
+    TimestampUtils.validateForLatestTimestamp(metaClient, instantTime);
```

Comment:
Just to confirm, is this happening within a table lock transaction?

Reply:
hey @kbuci : for 0.x branch, we are not taking a lock. but we are reloading the timeline while we do this validation which achieves the same purpose. in 1.x, commit time generation is w/n a lock. so, we should be good.